### PR TITLE
Add functions to handle notes

### PR DESCRIPTION
**Pull request by aurelienhess**
As there is currently no dedicated /notes/host endpoint this uses GET/PATCH on host objects directly. For consistency I'm artificially creating a similar requests.model.Reponse object to the one returned by requests.patch('{url}/tagging/host/{id}') as return when calling set_host_note(). 
Given that notes can be multiline, this is handled at the JSON level by encoding newlines in valid JSON.

### CODE REVIEW

File: modules/vectra.py [hunk 1]
```diff
@@ -267,6 +267,55 @@ def set_host_tags(self, host_id=None, tags=[], append=False):
...
+        :param host_id:
+        For consistency we return a requests.models.Response object
+        As we do not want to return the complete host body, we alter the response content
+        """
+        host = requests.get('{url}/hosts/{id}'.format(url=self.url, id=host_id), headers=self.headers, verify=self.verify)
+        host_note = host.json()['note']
+        # JSON encodes newline as \\n
+        # If there is no note, we encode as empty
+        host_note = host_note.replace('\n', '\\n') if host_note else ''
+        json_dict = '{{"status": "success", "host_id": "{}", "note":"{}"}}'.format(host_id,host_note)
```
Comment (HoneyBaked) [inline modules/vectra.py:285]: You may need to treat the note string as a unicode string.

"note": u"{}"

File: modules/vectra.py [hunk 2]
```diff
@@ -267,6 +267,55 @@ def set_host_tags(self, host_id=None, tags=[], append=False):
...
+        Set host note
+        :param host_id:
+        :param note: content of the note to set
+        :param append: overwrites existing note if set to False, appends if set to True
+        Set to empty note string to clear host note
+        """
+        if append and isinstance(note, str):
+            current_note = self.get_host_note(host_id=host_id).json()['note']
+            payload = {
+                "note": current_note + '\n' + note if current_note != '' else note
```
Comment (HoneyBaked) [inline modules/vectra.py:302]: We should make sure we treat it as a unicode string

```
"note":  u'{current_note}{separator}{new_note}'.format(current_note=current_note, separator=u'\n' if current_note else u'', new_note=new_note)
```

File: modules/vectra.py [hunk 3]
```diff
@@ -267,6 +267,55 @@ def set_host_tags(self, host_id=None, tags=[], append=False):
...
+            }
+        elif isinstance(note, str):
```
Comment (HoneyBaked) [inline modules/vectra.py:304]: isinstance(note, basestring)

File: modules/vectra.py [hunk 4]
```diff
@@ -267,6 +267,55 @@ def set_host_tags(self, host_id=None, tags=[], append=False):
...
+        For consistency we return a requests.models.Response object
+        As we do not want to return the complete host body, we alter the response content
+        """
+        host = requests.get('{url}/hosts/{id}'.format(url=self.url, id=host_id), headers=self.headers, verify=self.verify)
+        host_note = host.json()['note']
+        # JSON encodes newline as \\n
+        # If there is no note, we encode as empty
+        host_note = host_note.replace('\n', '\\n') if host_note else ''
+        json_dict = '{{"status": "success", "host_id": "{}", "note":"{}"}}'.format(host_id,host_note)
+        host._content = json_dict.encode('utf-8')
```
Comment (HoneyBaked) [inline modules/vectra.py:286]: You can use json.dumps to encode a dict to json

```
json_dict = {'status': 'success', 'host_id': str(host_id), 'note': host_note}
host._content = json.dumps(json_dict)
```

File: modules/vectra.py (hunk 2 above)
Comment (aurelienhess) [reply modules/vectra.py:302]: Aren't all strings treated as unicode by default in python 3?

Comment (aurelienhess) [reply modules/vectra.py:302]: I've run some tests and unicode characters seem to work seamlessly as well. The only thing bothering me is that we are HTML encoding some special chars in the notes field.

**Pull request closed without merging**